### PR TITLE
blob/gcsblob: add tests for OpenURL

### PR DESCRIPTION
**Pull request by vangent**
The `gcsblob.go` is just moving the function from being defined inline to being defined at the top level. It's more obvious if you "hide whitespace" in the diff.

Updates #675.

### CODE REVIEW

**Comment by vangent** [comment]
PTAL.

File: blob/gcsblob/gcsblob.go [hunk 1]
```diff
@@ -59,46 +59,48 @@ import (
...
+	opts := &Options{}
 
-		if keyPath := q["private_key_path"]; len(keyPath) > 0 {
-			pk, err := ioutil.ReadFile(keyPath[0])
-			if err != nil {
-				return nil, err
-			}
-			opts.PrivateKey = pk
-		}
+	if accessID := q["access_id"]; len(accessID) > 0 {
```
Comment (shantuo) [inline blob/gcsblob/gcsblob.go]: nit: `if accessID, ok := q["access_id"]; ok {`

File: blob/gcsblob/gcsblob.go [hunk 2]
```diff
@@ -59,46 +59,48 @@ import (
...
-		} else {
-			jsonCreds, err := ioutil.ReadFile(credPath[0])
-			if err != nil {
-				return nil, err
-			}
-			creds, err = google.CredentialsFromJSON(ctx, jsonCreds)
-			if err != nil {
-				return nil, err
-			}
+	if keyPath := q["private_key_path"]; len(keyPath) > 0 {
```
Comment (shantuo) [inline blob/gcsblob/gcsblob.go]: same

File: blob/gcsblob/gcsblob.go [hunk 3]
```diff
@@ -59,46 +59,48 @@ import (
...
+		pk, err := ioutil.ReadFile(keyPath[0])
+		if err != nil {
+			return nil, err
 		}
+		opts.PrivateKey = pk
+	}
 
-		client, err := gcp.NewHTTPClient(gcp.DefaultTransport(), gcp.CredentialsTokenSource(creds))
+	var creds *google.Credentials
+	if credPath := q["cred_path"]; len(credPath) == 0 {
```
Comment (shantuo) [inline blob/gcsblob/gcsblob.go]: same

File: blob/gcsblob/gcsblob.go (hunk 1 above)
Comment (vangent) [reply blob/gcsblob/gcsblob.go]: Hmm, I think checking that `len > 0` is more correct here; technically the map could have an `access_id` key with a 0-length slice value, and `ok` would be `true` but the access of `[0]` on the next line would panic. It's possible that `net.URL` would never produce a 0-length slice value, but I think its better to check the thing we actually care about.

**Pull request merged**